docs/database.md

```diff
@@ -0,0 +1,118 @@
+grant all privileges on st2020.* to 'admin'@'localhost';
+```
+
+Change the database and username appropriately and then update your `ST_DATABASE_URI` environment
+variable used when starting the Flask development server to match.
+
+## Accessing the database shell in Docker
+
+```bash
+docker-compose exec mysql mysql -uroot -p MYSQL_DATABASE
+```
+
+Enter your configured database root password when prompted.
+
+## Object-relational mapping
+
+SQL database represent data in the rows of relational tables. In contrast, programming languages
+like Python represent data as individual objects. Object-relational mapping (ORM) libraries bridge
+this gap by translating in both directions, creating an object-oriented abstraction for database
+access.
+
+[SQLAlchemy](https://docs.sqlalchemy.org/) is our choice of mapper for database access. We also use
+[Flask-SQLAlchemy](https://flask-sqlalchemy.palletsprojects.com/) to simplify using SQLAlchemy in
+a web context in a mostly transparent way: the SQLAlchemy database session is automatically created
+and destroyed with the Flask application context per request.
+
+[`models.py`](https://github.com/ccmbioinfo/stager/blob/master/flask/app/models.py) contains all
+the mappings for Stager.
+
+## Alembic migrations
+
+The above initialization only creates an empty database in MySQL. The application still needs to
+create the tables that correspond to its models and update the schemas for these tables if the
+models change during the course of development.
+
+We manage this through a the [Flask-Migrate](https://flask-migrate.readthedocs.io/) library, which
+integrates the [Alembic](https://alembic.sqlalchemy.org/) migration tool for SQLAlchemy with Flask.
+You should read Flask-Migrate's documentation if you need to do any work in this area.
+
+### Workflow (`models.py` is changed in a way that affects the SQL representation)
+
+Get a shell into the running app container (skip if using a virtualenv).
+
+```bash
+docker-compose exec app bash
+```
+
+Autogenerate an Alembic migration in [`migrations/versions`](https://github.com/ccmbioinfo/stager/tree/master/flask/migrations/versions).
+
+```bash
+flask db migrate -m "Commit message"
+```
+
+Verify the `upgrade` and `downgrade` functions in the file that was just created. Some changes are
+not automatically detected by Alembic, such as to enums, and column renames may not be represented
+ideally. Check constraints are also not applied and you may need to directly embed SQL code via
+`op.execute("MYSQL CODE HERE")` calls.
+
+Apply the migration and commit your changes to Git.
+
+```bash
+flask db upgrade
+```
+
+Because `flask db upgrade` is automatically called by the startup script [`./utils/run.sh`](https://github.com/ccmbioinfo/stager/blob/master/flask/utils/run.sh),
+you may experience problems when switching between branches that expect different database versions.
+Generally we just reset the database at this point, but you could avoid this by running
```

Comment:
by reset does this mean deleting ie.`rm -rf`-ing the mysql folder?

Reply:
That is usually how we do it and that's described in the Docker documentation under "Resetting bind mounts". It doesn't generally apply if you're running MySQL outside of Docker, where it might be easier to drop all the tables in the database. Should I clarify on both?

Reply:
that makes sense, I'm leaning towards clarifying on both